src/main/scala/chisel3/experimental/inlinetest/InlineTest.scala

```diff
@@ -95,6 +95,21 @@ object TestHarnessGenerator {
   */
 trait HasTests[M <: RawModule] { module: M =>
 
+  /** Whether inline tests will be elaborated as a top-level definition to the circuit. */
+  protected def elaborateTests: Boolean = true
+
+  private val builderContext = internal.Builder.captureContext()
+
+  private val thisModuleShouldElaborateTests =
+    builderContext.includeInlineTestsForModule.exists { glob =>
+      glob.replace("*", ".*").r.matches(desiredName)
```

Comment:
Please use `java.nio.file.PathMatcher` instead of converting globs to regex. Converting to regex has all kinds of issues (e.g. people can write a regex and it will "work"). `java.nio.file.PathMatcher` has native glob support, it was added for Paths but it just kind of works on raw Strings and it ships with the JVM so 🤷‍♀️ .

I used it for warning configuration: https://github.com/chipsalliance/chisel/blob/477900b9c72c89821f239fd800d495b077ddaf4e/core/src/main/scala/chisel3/internal/Error.scala#L160

The TLDR is you do something like `FileSystems.getDefault().getPathMatcher("glob:" + <your glob string>)` will give you a `PathMatcher` then `.matches(Paths.get(<arbitrary string>))` to check if the glob matches.